### PR TITLE
doc: Bluetooth: Mesh: update proxy solicitation docs

### PR DESCRIPTION
**Pull request by akredalen**
Updated related docs. Small fix in mesh/Kconfig and shell.rst.

### CODE REVIEW

**Review by jhedberg — DISMISSED** [review]

File: doc/connectivity/bluetooth/api/mesh/srpl_cli.rst [hunk 1]
```diff
@@ -3,7 +3,46 @@
...
+         .addr = 0x0004,
+         .net_idx = 0,
+         .send_ttl = BT_MESH_TTL_DEFAULT,
+      };
+
+      uint16_t range_start = 5;
+      uint8_t range_len = 20;
+
+      bt_mesh_sol_pdu_rpl_clear(&my_ctx, range_start, range_len,
+                                &start_rsp, &len_rsp);
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh/srpl_cli.rst:31]: Same as for On-Demand Private Proxy Client model. I'm not sure if we need such level of details. A user should be able to perceive this from the API description. If we have bad API description, we should fix it there.

File: doc/connectivity/bluetooth/api/mesh/od_cli.rst [hunk 2]
```diff
@@ -3,7 +3,53 @@
...
+      };
+
+      bt_mesh_od_priv_proxy_cli_get(&my_ctx, &state_rsp);
+
+and set a target's state using the :c:func:`bt_mesh_od_priv_proxy_cli_set` call:
+
+.. code-block:: C
+
+      uint8_t new_val = 100;
+      bt_mesh_od_priv_proxy_cli_set(&my_ctx, new_val, &state_rsp);
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh/od_cli.rst:38]: I'm not sure if we need such level of details. For such kind of functionality, it should be clear how to use the API. I think snippets are useful for more complex scenarios.

**Review by PavelVPV — CHANGES_REQUESTED** [review]
I think it would be good if we could elaborate the Proxy Solicitation feature in the documentation a little bit more by explaining the idea behind it (that any Proxy Client can solicit a Proxy Server to establish GATT connection at any time), the additionally required features (Private Node Identity), where the related models are instantiated (some can only be instantiated on Configuration Manager (On-Demand Private Proxy Client), some on any devices (Solicitation PDU RPL Client)), how Solicitation PDU is generated and how the solicitation replay protection list works (how and when entries are stored). 

No need to explain in detail how everything works, maybe one-two sentences per each aspect of the feature, but this should give enough information to a user to help building a general overview of the feature.

We have a page related to Proxy feature: https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/connectivity/bluetooth/api/mesh/proxy.rst. Probably, most of this information can be explained there. Models' specific details can be described in the corresponding models pages. Not sure if we need `od.rst` and `srpl.rst` after this, but lets see this once we add more description.

**Review by alxelax — COMMENTED** [review]
LGTM, but Pavel's comments seem fair enough. When they will be fixed, I'll approve also.

File: doc/connectivity/bluetooth/api/mesh/od_cli.rst [hunk 3]
```diff
@@ -3,7 +3,53 @@
 On-Demand Private Proxy Client
 ##############################
 
-On-Demand Private Proxy Client model is used to set and retrieve the On-Demand Private GATT Proxy state, and to adjust message transmission timeout.
+The On-Demand Private Proxy Client model is used to set and retrieve the On-Demand Private GATT Proxy state
+and to adjust message transmission timeout. The State defines how long a node will advertise Mesh Proxy Service with
```
[the remark this answers is not in the document]
Comment (akredalen) [reply doc/connectivity/bluetooth/api/mesh/od_cli.rst:7]: Should it be mentioned as a possible configuration, or is it unnecessary for this documentation? See the subsection further down.

**Comment by alxelax** [comment]
Hi @mia-ko, could you please review this PR?

**Review by PavelVPV — CHANGES_REQUESTED** [review]

File: doc/connectivity/bluetooth/api/mesh/proxy.rst [hunk 4]
```diff
@@ -9,6 +9,72 @@ mesh network through GATT. The Proxy feature is only compiled in if the
...
+
+The GATT Proxy state indicates if the Proxy feature is supported.
+
+Private Proxy
+*************
+
+A node supporting the :ref:`bluetooth_mesh_models_priv_beacon_srv` can advertise with
+Private Network Identity and Private Node Identity types, which is controlled by the
+:ref:`bluetooth_mesh_models_priv_beacon_cli`. Private Beacons provide privacy by periodically
+randomizing the beacon input data. By advertising with this set of identification types,
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh/proxy.rst:23]: Sentence `Private Beacons provide privacy by periodically randomizing the beacon input data.` is belongs to Private Beacons, but here we don't send beacons. The next sentence explains the need for Private types so I think this sentence can be removed.

File: doc/connectivity/bluetooth/api/mesh/proxy.rst [hunk 5]
```diff
@@ -9,6 +9,72 @@ mesh network through GATT. The Proxy feature is only compiled in if the
...
+the node is allowing its Proxy Client to connect to the network over GATT while maintaining the
+privacy of the network. This functionality is called Private Proxy. For the Private Proxy
+functionality to be supported, the Proxy feature, Private Network Identity advertising, and
+Mesh Private beacons must be supported by the node.
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh/proxy.rst:27]: I think the last 2 sentences aren't needed. What needs to be supported by the node and what Private Proxy gives is explained above. This is more like a spec statement, but for a user to enable the support of this feature it is enough to instantiate Private Beacon Server model and enable GATT Proxy support (which is explained in the section above).
```
This functionality is called Private Proxy. For the Private Proxy functionality to be supported, the Proxy feature, Private Network Identity advertising, and Mesh Private beacons must be supported by the node.
```

File: doc/connectivity/bluetooth/api/mesh/proxy.rst [hunk 6]
```diff
@@ -9,6 +9,72 @@ mesh network through GATT. The Proxy feature is only compiled in if the
...
 
+Nodes with the Proxy feature enabled can advertise with Network Identity and Node Identity,
+which is controlled by the :ref:`bluetooth_mesh_models_cfg_cli`.
+
+The GATT Proxy state indicates if the Proxy feature is supported.
+
+Private Proxy
+*************
+
+A node supporting the :ref:`bluetooth_mesh_models_priv_beacon_srv` can advertise with
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh/proxy.rst:20]:
```suggestion
A node supporting the Proxy feature and the :ref:`bluetooth_mesh_models_priv_beacon_srv` can advertise with
```

**Review by PavelVPV — CHANGES_REQUESTED** [review]

File: doc/connectivity/bluetooth/api/mesh/proxy.rst [hunk 7]
```diff
@@ -9,6 +9,50 @@ mesh network through GATT. The Proxy feature is only compiled in if the
...
+
+The Private GATT Proxy state indicates whether the Private Proxy functionality is supported.
+
+Proxy Solicitation
+******************
+
+In the case where both GATT Proxy and Private GATT Proxy states are disabled, the device cannot
+connect to the network over GATT. A device supporting the :ref:`bluetooth_mesh_od_srv` may however be
+solicited to advertise connectable advertising events without enabling the Private GATT Proxy state.
+To solicit the device, a Proxy Client may send a solicitation PDU by calling the :func:`bt_mesh_proxy_solicit`.
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh/proxy.rst:34]:
```suggestion
To solicit the device, a legacy device should send a solicitation PDU by calling the :func:`bt_mesh_proxy_solicit`.
```

File: doc/connectivity/bluetooth/api/mesh/proxy.rst [hunk 8]
```diff
@@ -9,6 +9,50 @@ mesh network through GATT. The Proxy feature is only compiled in if the
...
+option set.
+
+Solicitation PDUs are non-mesh, non-connectable, undirected advertising messages
+containing Proxy Solicitation UUID, encrypted with the network key of the network that the Client
+wants to connect to. The PDU contains the source address of the Client and a sequence number. The
+sequence number is maintained by the Client and is incremented for every new solicitation PDU sent.
+
+Each node supporting Solicitation PDU reception holds its own Solicitation Replay Protection List (SRPL).
+The SRPL protects the solicitation mechanism from replay attacks by storing solicitation sequence number (SSEQ)
+and solicitation source (SSRC) pairs of valid Solicitation PDUs processed by the node.
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh/proxy.rst:45]:
```suggestion
and solicitation source (SSRC) pairs of valid Solicitation PDUs processed by the node.
The delay between updating the SRPL and storing the change to the persistent storage is defined by :kconfig:option:`CONFIG_BT_MESH_RPL_STORE_TIMEOUT`.
```

File: doc/connectivity/bluetooth/api/mesh/proxy.rst [hunk 9]
```diff
@@ -9,6 +9,50 @@ mesh network through GATT. The Proxy feature is only compiled in if the
...
+The Solicitation PDU RPL Configuration models, :ref:`bluetooth_mesh_srpl_cli` and
+:ref:`bluetooth_mesh_srpl_srv`, provide the functionality of saving and clearing SRPL entries.
+The Client can clear a section of the SRPL on the target device by calling the :func:`bt_mesh_sol_pdu_rpl_clear`.
+Communication between the Solicitation PDU RPL Configuration Client and Server is encrypted using the application key, therefore,
+the Solicitation PDU RPL Configuration Client can be instantiated on any device in the network.
+
+When the server device receives the solicitation PDU and successfully authenticates it, it will start
+advertising connectable advertisements with the Private Network Identity type. The duration of the
+advertisement can be configured by the On-Demand Private Proxy Server model.
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh/proxy.rst:54]: Perhaps it should be client?
```suggestion
advertisement can be configured by the On-Demand Private Proxy client model.
```

**Review by PavelVPV — COMMENTED** [review]
@mia-ko could you please have a look now?

File: doc/connectivity/bluetooth/api/mesh/proxy.rst [hunk 10]
```diff
@@ -9,6 +9,50 @@ mesh network through GATT. The Proxy feature is only compiled in if the
...
+
+The GATT Proxy state indicates if the Proxy feature is supported.
+
+Private Proxy
+*************
+
+A node supporting the Proxy feature and the :ref:`bluetooth_mesh_models_priv_beacon_srv` model can advertise with
+Private Network Identity and Private Node Identity types, which is controlled by the
+:ref:`bluetooth_mesh_models_priv_beacon_cli`. By advertising with this set of identification types,
+the node is allowing its Proxy Client to connect to the network over GATT while maintaining the
```
[the remark this answers is not in the document]
Comment (akredalen) [reply doc/connectivity/bluetooth/api/mesh/proxy.rst:23]: "..the node is allowing its legacy device.." Is this correct? The legacy device is a potential node in a network, is it not? 
```suggestion
the legacy device is allowed to connect to the network over GATT while maintaining the
```

[the remark this answers is not in the document]
Comment (PavelVPV) [reply doc/connectivity/bluetooth/api/mesh/proxy.rst:23]: Ok, sounds weird. What about:
```
By advertising with this set of identification types, the node allows the legacy device to connect to the network over GATT while maintaining the privacy of the network.
```

File: doc/connectivity/bluetooth/api/mesh/srpl_srv.rst [hunk 11]
```diff
@@ -3,7 +3,31 @@
 Solicitation PDU RPL Configuration Server
 #########################################
 
-Solicitation PDU RPL Configuration Server model manages the solicitation RPL saved on the device.
+The Solicitation PDU RPL Configuration Server model is a foundation model defined by the Bluetooth
+mesh specification. The model is enabled if the node has the :ref:`bluetooth_mesh_od_srv` enabled.
+
+The Solicitation PDU RPL Configuration Server model was introduced in the Bluetooth Mesh Profile
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/srpl_srv.rst:9]:
```suggestion
The Solicitation PDU RPL Configuration Server model was introduced in the Bluetooth Mesh Protocol
```

File: doc/connectivity/bluetooth/api/mesh/srpl_srv.rst [hunk 12]
```diff
@@ -3,7 +3,31 @@
...
+Specification version 1.1, and manages the Solicitation Replay Protection List (SRPL) saved on the device.
+
+The SRPL is used to reject Solicitation PDUs that were already processed by a node. When a valid Solicitation PDU message is
+successfully processed by a node, the SSRC field and SSEQ field of the message is stored in the node's SRPL.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/srpl_srv.rst:13]:
```suggestion
The SRPL is used to reject Solicitation PDUs that are already processed by a node. When a valid Solicitation PDU message is
successfully processed by a node, the SSRC field and SSEQ field of the message are stored in the node's SRPL.
```

File: doc/connectivity/bluetooth/api/mesh/srpl_srv.rst [hunk 13]
```diff
@@ -3,7 +3,31 @@
...
+Configurations
+**************
+
+The Solicitation PDU RPL Configuration Server model has the following parameters available for configuration:
+
+Size of SRPL
+============
+
+The :kconfig:option:`CONFIG_BT_MESH_PROXY_SRPL_SIZE` sets the size of the SRPL. The list is used to determine if
+a received Solicitation PDU is valid.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/srpl_srv.rst:30]:
```suggestion
For the Solicitation PDU RPL Configuration Server model, the :kconfig:option:`CONFIG_BT_MESH_PROXY_SRPL_SIZE` option can be configured to set the size of the SRPL.
The SRPL is used to determine if a received Solicitation PDU is valid.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: doc/connectivity/bluetooth/api/mesh/proxy.rst [hunk 14]
```diff
@@ -9,6 +9,51 @@ mesh network through GATT. The Proxy feature is only compiled in if the
...
+wants to connect to. The PDU contains the source address of the legacy device and a sequence number. The
+sequence number is maintained by the legacy device and is incremented for every new solicitation PDU sent.
+
+Each node supporting Solicitation PDU reception holds its own Solicitation Replay Protection List (SRPL).
+The SRPL protects the solicitation mechanism from replay attacks by storing solicitation sequence number (SSEQ)
+and solicitation source (SSRC) pairs of valid Solicitation PDUs processed by the node. The delay between updating the
+SRPL and storing the change to the persistent storage is defined by :kconfig:option:`CONFIG_BT_MESH_RPL_STORE_TIMEOUT`.
+The Solicitation PDU RPL Configuration models, :ref:`bluetooth_mesh_srpl_cli` and
+:ref:`bluetooth_mesh_srpl_srv`, provide the functionality of saving and clearing SRPL entries.
+A node that supports the Solicitation PDU RPL Configuration Client model can clear a section of the SRPL on the target by calling the :func:`bt_mesh_sol_pdu_rpl_clear`.
```
[the remark this answers is not in the document]
Comment (akredalen) [reply doc/connectivity/bluetooth/api/mesh/proxy.rst:49]: It is meant in a general sense :)

**Review by alxelax — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]

**Pull request merged**